src/components/tasks/card/index.tsx

```diff
@@ -26,9 +29,28 @@ const Card: FC<Props> = ({
   const statusRedList = [BLOCKED];
   const statusNotOverDueList = [COMPLETED, VERIFIED, AVAILABLE];
   const cardDetails = content;
-  const [assigneeProfilePic, setAssigneeProfilePic] = useState(
-    `${process.env.NEXT_PUBLIC_GITHUB_IMAGE_URL}/${cardDetails.assignee}/img.png`,
-  );
+  const [assigneeProfilePic, setAssigneeProfilePic] = useState(DUMMY_PROFILE_PATH);
+
+  const memoizeAssignProfilePic = useCallback(async () => {
+    let isComponentMounted = true 
+    const {assignee} = cardDetails
+    if (assignee){
+      const response = await getCloudinaryImgURL(assignee,CLOUDINARY_IMAGE_CONFIGS)
+        if (isComponentMounted) {
+          if (response !== DUMMY_PROFILE_PATH) {
+            setAssigneeProfilePic(response)
+          }
+        }
+    }
+    return ()=>{
+      isComponentMounted = false
+    }
+  },[getCloudinaryImgURL,setAssigneeProfilePic])
```

Comment:
This seems like an anti-pattern. Could we dive deeper into not adding these safeguards manually please?

Reply:
yeah. I  have fixed it in my latest PR.